src/LinuxTracing/ThreadStateManagerTest.cpp

```diff
@@ -144,6 +163,8 @@ TEST(ThreadStateManager, TwoThreads) {
   EXPECT_EQ(slice->thread_state(), ThreadStateSlice::kRunnable);
   EXPECT_EQ(slice->duration_ns(), 100);
   EXPECT_EQ(slice->end_timestamp_ns(), 500);
+  EXPECT_EQ(slice->wakeup_pid(), kWasBlockedByPid1);
+  EXPECT_EQ(slice->wakeup_tid(), kWasBlockedByTid1);
```

Comment:
Maybe a stupid question, Are the thread that wakes up always the same than the one that blocks? @dpallotti

Reply:
I'm not sure I got the question. A thread blocks itself by doing something, e.g., requesting a resource. Another thread then unblocks it by making that resource available.